### ISSUE TITLE
Support of continuous benchmark and report

### ISSUE THREAD

**Issue by tonyhoo** [autogluon/autogluon#2262]
Support benchmark per merge in the actions workflow and report visualized results

Scikit-learn benchmark [page](https://scikit-learn.org/scikit-learn-benchmarks/) for reference

**Comment by gidler** [comment]
Since we had discussed [asv](https://asv.readthedocs.io/en/stable/index.html) several times related to this topic, and I had spent non-negligible time trying to get `asv` working in AutoGluon, I created this [PoC draft pull request](https://github.com/awslabs/autogluon/pull/2281) so if anyone else wants to investigate, they can build off the work I've already put into it.

The pull request is not intended to be production ready. It provides an example `asv.config.json` which gets `asv` working in the AutoGluon repo and provides a simple benchmark file with ways we could benchmark `TabularPredictor` using the `asv` approach.

After working on this, I don't believe `asv` is the exact tool we were looking for for tracking benchmark performance over time, but we will make that design decision when we have codified requirements and looked at alternatives.
